File: firmipbo/src/java/it/csi/firmip/firmipbo/business/service/TipoDocumentoService.java
/*
* SPDX-FileCopyrightText: (C) Copyright 2023 Consiglio Regionale Piemonte
*
* SPDX-License-Identifier: EUPL-1.2 */
package it.csi.firmip.firmipbo.business.service;

import java.util.List;

import it.csi.firmip.firmipbo.dto.TipoDocumento;

/**
 * Metodi di business relativi ai tipi di documento
 *
 * @see TipoDocumento
 *
 */
public interface TipoDocumentoService {

	public List<TipoDocumento> getElencoTipoDocumento();
	public TipoDocumento getTipoDocumentoByTipo(String tipo);
	
}

File: firmipbowcl/src/app/services/tipo-documento.service.ts
/*
* SPDX-FileCopyrightText: (C) Copyright 2023 Consiglio Regionale Piemonte
*
* SPDX-License-Identifier: EUPL-1.2 */
import { Injectable } from '@angular/core';
import {ResourceHttpService} from "./resource-http.service";
import {Iniziativa, TipoDocumento} from "../model/firmipbo-data-model";
import {HttpClient} from "@angular/common/http";

@Injectable({
  providedIn: 'root'
})
export class TipoDocumentoService extends ResourceHttpService<TipoDocumento> {

  constructor(protected override httpClient: HttpClient
  ) {super(httpClient);

  }

  getResourceUrl(): string {
    return 'tipi-documento';
  }

}
